Libraries/LibWeb/HTML/SessionHistoryTraversalQueue.cpp

```diff
@@ -27,10 +27,16 @@ void SessionHistoryTraversalQueueEntry::visit_edges(JS::Cell::Visitor& visitor)
 SessionHistoryTraversalQueue::SessionHistoryTraversalQueue()
 {
     m_timer = Core::Timer::create_single_shot(0, [this] {
+        while (m_higher_priority_queue.size() > 0) {
+            auto entry = m_higher_priority_queue.take_first();
+            entry->execute_steps();
+        }
+
         if (m_is_task_running && m_queue.size() > 0) {
             m_timer->start();
             return;
         }
```

Comment:
Aren't you bypassing the `m_is_task_running` logic with the higher priority tasks now?

Reply:
OK, my point was that `m_is_task_running` guards against iterating over the queue again while we're still processing it, and we have no such protection for the new, higher priority queue.